library/src/nvidia_detail/hipsolver.cpp

```diff
@@ -89,6 +89,49 @@ catch(...)
     return hipsolver::exception2hip_status();
 }
 
+hipsolverStatus_t hipsolverSetDeterministicMode(hipsolverHandle_t            handle,
+                                                hipsolverDeterministicMode_t mode)
+try
+{
+#if(CUDART_VERSION >= 12020)
+    if(!handle)
+        return HIPSOLVER_STATUS_NOT_INITIALIZED;
+
+    return hipsolver::cuda2hip_status(cusolverSetDeterministicMode(
+        (cusolverDnHandle_t)handle, hipsolver::hip2cuda_deterministic(mode)));
+#else
+    return HIPSOLVER_STATUS_NOT_SUPPORTED;
+#endif
```

Comment:
We may want to ask around whether the preference is to return `HIPSOLVER_STATUS_NOT_SUPPORTED` or to drop the function from the API entirely. This is fine for now, but that's a discussion we should pencil in for later.